### PR TITLE
changes to be implemented

### PR DESCRIPTION
**Pull request by cwverica**
-Restructuring of generateStoryMarkup
-Consolidation of all API handling to models.py

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ function hidePageComponents() {
 /** Overall function to kick off the app. */
 
 async function start() {
-  console.debug("start");
+  // console.debug("start");
```
Comment (montrealist) [inline js/main.js]: Try to remove these if not needed. :)

Comment (montrealist) [reply js/main.js]: Also, down below, `if (currentUser)` - what happens if no currentUser exists? Maybe you're planning to address in another PR, just wanted to point it out.

File: js/models.js [hunk 2]
```diff
@@ -104,6 +104,31 @@ class StoryList {
       /*
     old code, might be useful at some point */
   }
+
+  /* Recieves a user obj and a storyId
+  *   -verifies story belongs to user
+  *   -deletes story from api
+  */
+  async deleteStoryFromList(user, storyId){
+    const belongsToUser = user.ownStories.some((story) => (story.storyId == storyId));
```
Comment (montrealist) [inline js/models.js:113]: Nice usage of `some()`! 😎

File: js/models.js [hunk 3]
```diff
@@ -104,6 +104,31 @@ class StoryList {
...
+    if(belongsToUser){
+      try{
+        await axios({
```
Comment (montrealist) [inline js/models.js]: Minor but would be nice to check that delete succeeded:
```
const res = await axios.delete(...);
if(res.status === 200) {...}
```

File: js/user.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ let currentUser;
 /** Handle login form submission. If login ok, sets up the user instance */
 
 async function login(evt) {
```
Comment (montrealist) [inline js/user.js]: Would be good to rename to something like `loginForm` - I got a bit confused at first why there were 2 `login()` functions - on here, one in model.

File: js/models.js [hunk 5]
```diff
@@ -104,6 +104,31 @@ class StoryList {
...
+          method: "DELETE",
+          data: {
+            "token": user.loginToken
+          }
+        });
+        currentUser = await User.updateUser();
+        displayMyStories();
+      }
+      catch (e){
+        console.error("Failed to delete story", e)
```
Comment (montrealist) [inline js/models.js:132]: Could return `false` here (or `true` above) to check for it in `deleteStory()` and show feedback to user.

File: js/models.js [hunk 6]
```diff
@@ -104,6 +104,31 @@ class StoryList {
...
+      try{
+        await axios({
+          url: `${BASE_URL}/stories/${storyId}`,
+          method: "DELETE",
+          data: {
+            "token": user.loginToken
+          }
+        });
+        currentUser = await User.updateUser();
+        displayMyStories();
```
Comment (montrealist) [inline js/models.js]: `displayMyStories()`: models should never call functions from outside. Their responsibility is only to fetch/send data from/to the persistence layer and return the result of each operation to whomever called it (either data or error/success status). Please refactor to remove this call.

**Review by montrealist — CHANGES_REQUESTED** [review]
Mostly minor except the `displayMyStories()` comment - that's why I chose 'request changes' instead of 'approve'. You can push your updates to the same branch on Github (dev) and they should become part of this pull request.

File: js/main.js (hunk 1 above)
Comment (cwverica) [reply js/main.js]: > Also, down below, `if (currentUser)` - what happens if no currentUser exists? Maybe you're planning to address in another PR, just wanted to point it out.

If you're talking about line 51, that's just a check to see if there is a user to update the nav bar and load in starred-favorites. It's already loaded the page, so if there is no user, it doesn't need to make a change.

edited: for clarity, and also I had the wrong line number (I had already made changes, and was referencing my forked code)

File: js/models.js (hunk 2 above)
Comment (cwverica) [reply js/models.js:113]: Thanks, I would rather do that then check indexOf

**Review by montrealist — APPROVED** [review]

**Comment by montrealist** [comment]
LGTM!

**Pull request merged**